shared/roughtime/roughtime.go

```diff
@@ -40,20 +41,22 @@ var offsetsRejected = promauto.NewCounter(prometheus.CounterOpts{
 })
 
 func init() {
-	recalibrateRoughtime()
-	go func() {
-		for {
-			wait := RecalibrationInterval
-			// recalibrate every minute if there is a large skew.
-			if offset > 2*time.Second {
-				wait = 1 * time.Minute
+	if flags.EnableRoughtime.IsSet() {
```

Comment:
I think you're going to need a delay before this. This init function will run before any flags are read.